web3.js/test/connection.test.ts

```diff
@@ -4329,5 +4333,121 @@ describe('Connection', function () {
         expect(lookupTableAccount.state.authority).to.be.undefined;
       }
     });
+
+    it('sendRawTransaction with v0 transaction', async () => {
+      const payer = Keypair.generate();
+
+      await helpers.airdrop({
+        connection,
+        address: payer.publicKey,
+        amount: 10 * LAMPORTS_PER_SOL,
+      });
+
+      const lookupTableAddresses = [Keypair.generate().publicKey];
+      const recentSlot = await connection.getSlot('finalized');
+      const [createIx, lookupTableKey] =
+        AddressLookupTableProgram.createLookupTable({
+          recentSlot,
+          payer: payer.publicKey,
+          authority: payer.publicKey,
+        });
+
+      // create, extend, and fetch lookup table
+      {
+        const transaction = new Transaction().add(createIx).add(
+          AddressLookupTableProgram.extendLookupTable({
+            lookupTable: lookupTableKey,
+            addresses: lookupTableAddresses,
+            authority: payer.publicKey,
+            payer: payer.publicKey,
+          }),
+        );
+        await helpers.processTransaction({
+          connection,
+          transaction,
+          signers: [payer],
+          commitment: 'processed',
+        });
+
+        const lookupTableResponse = await connection.getAddressLookupTable(
+          lookupTableKey,
+          {
+            commitment: 'processed',
+          },
+        );
+        const lookupTableAccount = lookupTableResponse.value;
+        if (!lookupTableAccount) {
+          expect(lookupTableAccount).to.be.ok;
+          return;
+        }
+
+        // eslint-disable-next-line no-constant-condition
+        while (true) {
+          const latestSlot = await connection.getSlot('processed');
+          if (latestSlot > lookupTableAccount.state.lastExtendedSlot) {
+            break;
+          } else {
+            console.log('Waiting for next slot...');
+            await sleep(500);
+          }
+        }
+      }
+
+      // create, serialize, send and confirm versioned transaction
+      {
+        const {blockhash, lastValidBlockHeight} =
+          await connection.getLatestBlockhash();
+        const transferIxData = encodeData(SYSTEM_INSTRUCTION_LAYOUTS.Transfer, {
+          lamports: BigInt(LAMPORTS_PER_SOL),
+        });
+        const transaction = new VersionedTransaction(
+          new MessageV0({
+            header: {
```

Comment:
So using the new compilation to build this transaction would work like this?
```
const transaction = new VersionedTransaction(
    MessageV0.compile({
        payerKey: payer.publicKey,
        instructions: [transferIx],
        recentBlockhash: blockhash,
        addressLookupTableAccounts: [lookupTableAccount],
    })
);
```
Sounds good to me.

Maybe this test could send a second transaction that was compiled that way? Possibly add a handful of testcases where the MessageV0.compile() output is compared to an expected result?